bosh-stemcell/spec/stemcells/fips_spec.rb

```diff
@@ -49,12 +55,35 @@
 
     let(:dpkg_list_ubuntu) { File.readlines(spec_asset('dpkg-list-ubuntu-jammy.txt')).map(&:chop) }
     let(:dpkg_list_fips_ubuntu) { File.readlines(spec_asset('dpkg-list-ubuntu-jammy-fips.txt')).map(&:chop) }
+    let(:dpkg_list_aws_fips_ubuntu) { File.readlines(spec_asset('dpkg-list-ubuntu-jammy-aws-fips.txt')).map(&:chop) }
     let(:dpkg_list_google_ubuntu) { File.readlines(spec_asset('dpkg-list-ubuntu-jammy-google-additions.txt')).map(&:chop) }
     let(:dpkg_list_vsphere_ubuntu) { File.readlines(spec_asset('dpkg-list-ubuntu-jammy-vsphere-additions.txt')).map(&:chop) }
     let(:dpkg_list_azure_ubuntu) { File.readlines(spec_asset('dpkg-list-ubuntu-jammy-azure-additions.txt')).map(&:chop) }
     let(:dpkg_list_cloudstack_ubuntu) { File.readlines(spec_asset('dpkg-list-ubuntu-jammy-cloudstack-additions.txt')).map(&:chop) }
     let(:dpkg_list_softlayer_ubuntu) { File.readlines(spec_asset('dpkg-list-ubuntu-jammy-softlayer-additions.txt')).map(&:chop) }
 
+    describe command(dpkg_list_packages), {
+      exclude_on_alicloud: true,
+      exclude_on_cloudstack: true,
+      exclude_on_vcloud: true,
+      exclude_on_vsphere: true,
+      exclude_on_warden: true,
+      exclude_on_azure: true,
+      exclude_on_openstack: true,
+      exclude_on_softlayer: true,
+    } do
+      let(:infrastructure) { ENV['STEMCELL_INFRASTRUCTURE'] }
+      let(:use_iaas_kernel) { ENV['UBUNTU_FIPS_USE_IAAS_KERNEL'] }
+
+      it 'contains only the base set of packages plus aws-specific kernel packages' do
+        skip "Test skipped due to non-aws kernel" if (use_iaas_kernel && infrastructure != 'aws')
```

Comment:
Could this line be changed to:
```
skip "Test skipped due to generic kernel" unless use_iaas_kernel
```
The infrastructure check should not be needed because the excludes.